sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringExpressions.scala

```diff
@@ -343,47 +343,71 @@ case class FindInSet(left: Expression, right: Expression) extends BinaryExpressi
 }
 
 /**
- * A function that trim the spaces from both ends for the specified string.
+ * A function that trims the given character (by default space) from both ends for given string.
  */
-case class StringTrim(child: Expression)
-  extends UnaryExpression with String2StringExpression {
+case class StringTrim(left: Expression, right: Expression)
```

Comment:
Nit: I am wondering if we can use more meaningful name, like `expr` and `trimChar` for constructor arguments.
And also add `def left = expr` and `def right = trimChar`.